### ISSUE TITLE
[Need discussion] Wrong Message Displayed When User Has Been Locked by Backend

### ISSUE THREAD

**Issue by giancorderoortiz** [SAP/spartacus#3690]
**Setup:** 
1. In backend, through backoffice, set max login attempts. (for example, 3)
1.a. See https://help.sap.com/viewer/d0224eca81e249cb821f2cdf45a82ace/1905/en-US/28cfbf93784640d09929ee4e9b97ba1e.html?q=brute%20force%20prevention

**Steps to Reproduce:** 
1. Navigate to Spartacus storefront. 
2. Register a user. (i.e. john.doe)
3. Logout. 
4. Return to login page. 
5. Attempt to login with invalid password. 
6. Repeat 2 times. 

**Expected Results:**
6. In the third attempt, the error message should indicate account has been locked, and user should contact Administrator. 

**Actual Results:**
6. Error message says "User is disabled, Please login again". This is in fact, not helpful, given that account is locked and subsequent efforts, even if providing right password will fail. 

(pending imgs)

**Comment by hackergil** [comment]
Let's check if the OCC API gives us a way to know that the account has been locked

**Comment by WeizhengSap** [comment]
Set `Max brute force login attempts` to 3. The user is disabled when the 4th login (not the 3rd). So, after try to login 4 times, I see this:
![Screen Shot 2019-07-18 at 3 40 44 PM](https://user-images.githubusercontent.com/44440575/61486917-a3253000-a972-11e9-836b-c1f4a36e808f.png)

**Comment by giancorderoortiz** [comment]
The message "User is disabled. Please login again" is not helpful. 
The user cannot login again until account is unblocked by Administrator. 
It should say "User is disabled. Please contact Administrator". 
(I'll update ticket description)

**Comment by WeizhengSap** [comment]
The error message is received from backend. So, need to create a new ticket to occ team.

**Comment by giancorderoortiz** [comment]
There are cases when there is a need to improve the message received from backend. A broader strategy is needed in order to avoid spaghetti code. 
Given the longer period of backend releases, frontend should take a proactive approach so that end-user benefits from error information that is actually useful. 
Please leave open until further discussions are held.

**Comment by Xymmer** [comment]
We can intercept this using localization
Bill to come back with a message and bring into a sprint

**Comment by giancorderoortiz** [comment]
in fact, what is received from backend is: 
{
  "error" : "invalid_grant",
  "error_description" : "User is disabled"
}
"Please login again" is added afterwards.

**Comment by giancorderoortiz** [comment]
![Screen Shot 2019-07-29 at 5 31 53 PM](https://user-images.githubusercontent.com/46171897/62085256-1e1b0000-b229-11e9-848a-2c5376623423.png)

**Comment by giancorderoortiz** [comment]
@zebercut , @Xymmer can this be fixed for 3.0? This minor security/usability issue has been opened for more than 1 year.

**Comment by dydome** [comment]
@WeizhengSap Is there any possibility to add any specific code for this kind of error - `disabled user'. We have to somehow know about this specific error as for now there is a condition to check the error field and additional code will help us to change the text.

**Comment by giancorderoortiz** [comment]
Still an issue with 3.0 and develop branch (June 11). 
Should be fixed. 
Error message doesn't make sense

**Comment by giancorderoortiz** [comment]
@Xymmer, @KateChuen should this be handled for 5.0 ?
